var/spack/repos/builtin/packages/openloops/package.py

```diff
@@ -0,0 +1,123 @@
+        "ppllll", "ppllll2", "ppllll2_nf5",
+        "ppllll2_onlyh", "ppllll_ew", "ppllllj", "ppllllj2", "ppllllj2_nf5",
+        "ppllllj2_nf5_notridr", "ppllllj2_nf5_sr",
+        "ppllllj2_onlyh", "ppllnnjj_ew", "ppllnnjj_vbs", "pplnajj", "ppvv",
+        "ppvv2", "ppvv_ew", "ppvvj", "ppvvj2",
+        "ppvvj_ew", "ppwajj", "ppwwjj", "ppzajj", "ppzwj_ew", "ppzwjj",
+        "ppzzjj", "ppajj", "ppajj2", "ppajj_ew", "ppajjj",
+        "ppllj", "ppllj2", "ppllj_ew", "ppllj_nf5", "pplljj", "pplljj_ew",
+        "pplljjj", "pplnj_ckm", "pplnjj", "pplnjj_ckm",
+        "pplnjj_ew", "pplnjjj", "ppnnjj_ew", "ppnnjjj", "ppvj", "ppvj2",
+        "ppvj_ew", "ppwj_ckm", "ppwjj", "ppwjj_ckm",
+        "ppwjj_ew", "ppwjjj", "ppzjj", "ppzjj_ew", "ppzjjj", "pphtt",
+        "pphtt_ew", "pphttj", "pphlltt", "pphll", "pphll2",
+        "pphll_ew", "pphllj", "pphllj2", "pphllj_ew", "pphlljj", "pphlljj_top",
+        "pphlnj_ckm", "pphlnjj", "pphv", "pphv_ew",
+        "pphwjj", "pphz2", "pphzj2", "pphzjj", "pphhtt", "pphhv", "pphhh2",
+        "heftpphh", "heftpphhj", "heftpphhjj", "pphh2",
+        "pphhj2", "pphhjj2", "pphhjj_vbf", "bbhj", "heftpphj", "heftpphjj",
+        "heftpphjjj", "pphbb", "pphbbj", "pphj2",
+        "pphjj2", "pphjj_vbf", "pphjj_vbf_ew", "pphjjj2", "eetttt", "eettttj",
+        "eellllbb", "eett", "eett_ew", "eettj",
+        "eettjj", "eevtt", "eevttj", "eevttjj", "eevvtt", "eevvttj",
+        "eellll_ew", "eevv_ew", "eevvjj", "eell_ew", "eevjj",
+        "eehtt", "eehttj", "eehll_ew", "eehvtt", "eehhtt", "heftppllj",
+        "heftpplljj", "heftpplljjj")
+
+    variant('compile_extra', default=False,
+            description='Compile real radiation tree amplitudes')
+    variant('processes', description='Processes to install. See https://' +
+                                     'openloops.hepforge.org/process_' +
+                                     'library.php?repo=public for details',
+            values=disjoint_sets(('all.coll',), ('lhc.coll',), ('lcg.coll',),
+                                 all_processes).with_default('lhc.coll'))
+
+    variant('num_jobs', description='Number of parallel jobs to run. '  +
+                                    'Set to 1 if compiling a large number' +
+                                    'of processes (e.g. lcg.coll)', default=0)
+    depends_on('python', type=("build", "run"))
+
+    phases = ['configure', 'build', 'build_processes', 'install']
+
+    def configure(self, spec, prefix):
+        spack_env = ('PATH LD_LIBRARY_PATH CPATH C_INCLUDE_PATH' +
+                     'CPLUS_INCLUDE_PATH INTEL_LICENSE_FILE').split()
+        for k in env.keys():
+            if k.startswith('SPACK_'):
+                spack_env.append(k)
+
+        spack_env = ' '.join(spack_env)
+        is_intel = self.spec.satisfies('%intel')
+        njobs = self.spec.variants['num_jobs'].value
+
+        with open('openloops.cfg', 'w') as f:
+            print('[OpenLoops]', file=f)
```

Comment:
```suggestion
            f.write('[OpenLoops]\n')
```

Reply:
I was assuming that Spack targets python3 (as far as I can see, when possible it chooses python3), and "behind the scenes" does `from __future__ import ...`.

Reply:
Nope. We equally target Python 2.6+ and 3.5+ with no magic future imports. Also, https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files still recommends using `f.write` even in Python 3.8.